tutorial/support-svg-sprite.md

```diff
@@ -0,0 +1,149 @@
+
+**DOM-like and style-able** - SVG images are like code, so this means it can be navigated like a DOM element and also styled, although some properties have different names eg(you might want to use fill instead of color). It can also be styled with CSS.
+
+**Editable** - Since SVGs are DOM-like, SVG’s can be created and edited
+
+## What is SVG's in JS?
+
+Like HTML, SVGs are represented using the [Document Object Model](https://en.wikipedia.org/wiki/Document_Object_Model "Document Object Model") (DOM) and so can be manipulated with Javascript relatively easily. 
+##### Problem.
+There are SVG's embedded in JS for icons / logos / etc. These cause an impact on performance because:
+- The size of the JS increases with additions and entail cost over the network
+- Render blocking
+- Downloaded JS is parsed affecting FCP & FMP.
+
+##### Solution
+SVG's should be made external via webpack / plugins to load them lazy via a network call or DOM reference. Thus we are using `SVG sprite loader` (Webpack loader for creating SVG sprites.) 
+
+## Why SVG sprite loader?
+SVG Sprites are rendered and injected in pages automatically, you just refer to images via `<svg><use xlink:href="#id"></use></svg>`. Also it can render sprites on server or in browser manually.  
+
+## Set up [SVG sprite loader](https://www.npmjs.com/package/svg-sprite-loader "SVG sprite loader") for your app
+
+There are few steps you need to follow to set up SVG sprite loader:
+
+1. The first thing you need to install the package `npm install svg-sprite-loader -D`
+2. Go to `webpack.config.js` and modify the webpack config to support svg-sprite-loader.
+
+```javascript
+const webpackConfig = require("@quintype/build/config/webpack");
+const path = require("path");
+
+const SpriteLoaderPlugin = require("svg-sprite-loader/plugin");
+const svgSprite = {
+  test: /\.svg$/,
+  loader: "svg-sprite-loader",
+  options: {
+    extract: true,
+    publicPath: "/",
+    symbolId: filePath => {
+      return path
+        .basename(filePath)
+        .replace(".svg", "")
+        .toLowerCase(); // 
+    },
+    spriteFilename: "sprite.svg",
+    esModule: false
+  }
+};
+
+webpackConfig.module.rules.push(svgSprite);
+webpackConfig.module.rules.find(rule => rule.loader === "file-loader").exclude = [/app\/assets\/icons\/[a-z-]+\.svg$/];
+
+const svgPlugin = () =>
+  new SpriteLoaderPlugin({
+    plainSprite: true
+  });
+
+webpackConfig.plugins.push(svgPlugin());
+
+module.exports = {
+  ...webpackConfig
+};
+
+```
+Once you compile(build) the webpack, `sprite.svg` will be generated and available in the public folder.
+
+3 .  Runtime configuration 
+When you require an image, the loader transforms it to SVG <symbol>, adds it to the special sprite storage and returns class instance that represents symbol. It contains id, viewBox and content (id, viewBox and url in extract mode) fields and can later be used for referencing the sprite image.
+
+ - Add all the svg icons inside `/assets/icons/` folder.
```

Comment:
This comes without any heading

Reply:
the first point in   Runtime configuration